lib/archive_importer/post_importer.rb

```diff
@@ -0,0 +1,34 @@
+# frozen_string_literal: true
+
+class ArchiveImporter
+  class PostImporter < OwnEntityImporter
+    include Diaspora::Logging
+
+    def substitute_author
+      super
+      return unless entity_type == "status_message"
+      entity_data["photos"].each do |photo|
+        photo["entity_data"]["author"] = user.diaspora_handle
+      end
+    end
+
+    def import
+      super
+      import_subscriptions if persisted_object
+    end
+
+    private
+
+    def import_subscriptions
+      object.fetch("subscribed_users_ids", []).each do |diaspora_id|
+        begin
+          person = Person.find_or_fetch_by_identifier(diaspora_id)
+          next if person.closed_account?
+          person = person.account_migration.newest_person unless person.account_migration.nil?
+          persisted_object.participations.find_or_create_by(author: person) if person
```

Comment:
Why not create a participation as a subscription before we have a better structure for it? Participations work as subscriptions and I think when you create the new structure we'll have to convert participations to new subscriptions anyway. Participations are not exactly subscriptions, because they also work as notification toggle in terms of UX, but since it is only for development version, I think it is fine. And by creating participations we will not lose the information about subscriptions for accounts which were imported with the development code.

Reply:
For public posts you can create a participation instead of a subscription, but you need to select a random person from the pod in `subscribed_pods_uris`. And yes, I'll do a migration of participations for public posts (that part is already done, but I need more time for testing and edge-cases). So if you want to create participations for public posts as a workaround, feel free to do so.

For private posts (as you do it here) this workaround doesn't work.

Reply:
> For private posts (as you do it here) this workaround doesn't work.

Is it because there is no `AspectVisibility` for the person?

Reply:
Yes, the subscriptions for private posts are tracked via `AspectVisibility` at the moment, and you can't create an `AspectVisibility` from the `subscribed_users_ids`.

Reply:
I see.

Is it okay if I remove the line 28 and place a `# TODO` comment there, but leave everything else in this method? The fetching code is relevant and I think it will work for the new structure the same way. It will be redundant for now, but I don't think it is a problem.

Reply:
Yes, as written before, a `# TODO` is OK, and you're right, the other lines are needed anyway later, so you can leave them :+1: